flink-filesystems/flink-hadoop-fs/src/main/java/org/apache/flink/runtime/util/HadoopUtils.java

```diff
@@ -77,47 +75,18 @@ public static Configuration getHadoopConfiguration(
 
         for (String possibleHadoopConfPath : possibleHadoopConfPaths) {
             if (possibleHadoopConfPath != null) {
-                foundHadoopConfiguration = addHadoopConfIfFound(result, possibleHadoopConfPath);
+                foundHadoopConfiguration |= addHadoopConfIfFound(result, possibleHadoopConfPath);
```

Comment:
This is a simple boolean arithmetic in order not to overwrite the first round true with a second round false.